rank-info/src/fetcher.rs

```diff
@@ -0,0 +1,260 @@
+                            rating_update_count: rank_resp.rating_update_count,
+                            rating_change: rating_change,
+                            rank_change: rank_change as i32,
+                        });
+
+                        // debug logs
+                        let test = rank_data.current_rank.clone().unwrap();
+                        tracing::info!(target: Log::SlippiOnline, "rank: {0}", test.rank);
+                        tracing::info!(target: Log::SlippiOnline, "rating_ordinal: {0}", test.rating_ordinal);
+                        tracing::info!(target: Log::SlippiOnline, "global_placing: {0}", test.global_placing);
+                        tracing::info!(target: Log::SlippiOnline, "regional_placing: {0}", test.regional_placing);
+                        tracing::info!(target: Log::SlippiOnline, "rating_update_count: {0}", test.rating_update_count);
+
+                        Ok(RankInfo {
+                            rank: 0,
+                            rating_ordinal: 0.0,
+                            global_placing: 0,
+                            regional_placing: 0,
+                            rating_update_count: 0,
+                            rating_change: rating_change,
+                            rank_change: rank_change as i32,
+                        })
+                    },
+                    Err(_err) => Err(GetRankErrorKind::NotSuccessful("Failed to parse rank struct".to_owned())),
+                }
+            },
+            Err(err) => Err(err),
+        }
+    }
+}
+
+pub fn run(fetcher: RankInfoFetcher, receiver: Receiver<Message>) {
+    loop {
+        match receiver.recv() {
+            Ok(Message::FetchRank) => {
+                let connect_code = fetcher.user_manager.get(|user| user.connect_code.clone());
+                let _ = fetcher.fetch_user_rank(&connect_code);
+            },
+
+            Ok(Message::RankFetcherDropped) => {
+                tracing::info!(target: Log::SlippiOnline, "Rank fetcher thread dropped");
+            },
+
+            Err(error) => {
+                tracing::error!(
+                    target: Log::SlippiOnline,
+                    ?error,
+                    "Failed to receive Message, thread will exit"
+                );
+
+                break;
+            },
+        }
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum GetRankErrorKind {
+    #[error("GetRankErrorKind Network")]
+    Net(slippi_gg_api::Error),
+
+    #[error("GetRankErrorKind JSON")]
+    JSON(serde_json::Error),
+
+    #[error("GetRankErrorKind GraphQL")]
+    GraphQL(String),
+
+    #[error("GetRankErrorKind NotSuccessful")]
+    NotSuccessful(String),
+}
+
+#[derive(Debug, Clone, Default, serde::Deserialize)]
+pub struct RankInfoAPIResponse {
+    #[serde(alias = "ratingOrdinal")]
+    pub rating_ordinal: f32,
+
+    #[serde(alias = "ratingUpdateCount")]
+    pub rating_update_count: u32,
+
+    #[serde(alias = "dailyGlobalPlacement", default)]

```

Comment:
You might not need the `default` in the `#[serde()]` derive here, since `Default` on the entire struct will just do `None` for this value (and for `daily_regional_placement`).

Reply:
(and `serde` will do `None` for an Option as a default anyway)